crates/fluvio-hub-util/src/fvm/api/client.rs

```diff
@@ -89,20 +108,36 @@ mod tests {
     }
 
     #[test]
-    fn builds_url_for_fetching_pkgsets() {
+    fn builds_urls_for_fetching_pkgsets() {
+        // Scenario: Using Stable Channel
+
         let client = Client::new("https://hub.infinyon.cloud").unwrap();
         let url = client
             .make_fetch_package_set_url(&Channel::Stable, "arm-unknown-linux-gnueabihf")
             .unwrap();
 
         assert_eq!(
             url.as_str(),
-            "https://hub.infinyon.cloud/hub/v1/fvm/pkgset/stable?arch=arm-unknown-linux-gnueabihf"
+            "https://hub.infinyon.cloud/hub/v1/fvm/pkgset/stable?arch=arm-unknown-linux-gnueabihf",
+            "failed on Scenario Using Stable Channel"
         );
-    }
 
-    #[test]
-    fn builds_url_for_fetching_pkgsets_on_version() {
+        // Scenario: Using Tag
```

Comment:
Centralized `builds_url_...` tests because using `set_var` is not thread safe, so each scenario is centralized in a single test with multiple assertions.